libraries/Microsoft.Bot.Builder.Prompts/ConfirmPrompt.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+using Microsoft.Recognizers.Text;
+using Microsoft.Recognizers.Text.DateTime;
+using Microsoft.Recognizers.Text.Number;
+using Microsoft.Recognizers.Text.NumberWithUnit;
+using static Microsoft.Bot.Builder.Prompts.PromptValidatorEx;
+
+namespace Microsoft.Bot.Builder.Prompts
+{
+    public class ConfirmResult : RecognitionResult
+    {
+        public ConfirmResult() { }
+
+        public bool Confirmation { get; set; }
+
+        public string Text { get; set; }
+    }
+
+    /// <summary>
+    /// ConfirmPrompt recognizes confrimation expressions as bool 
+    /// </summary>
+    public class ConfirmPrompt : BasePrompt<ConfirmResult>
+    {
+        // private IModel _model;
+        private Regex yes = new Regex(@"(\byes\b|\byep\b|\bok\b|\byessir\b|\bconfirm\b|^y$)", RegexOptions.IgnoreCase);
+        private Regex no = new Regex(@"(\bno\b|\bnope\b|\bnosir\b|\bcancel\b|^n$)", RegexOptions.IgnoreCase);
+
+        public ConfirmPrompt(string culture, PromptValidator<ConfirmResult> validator = null)
+            : base(validator)
+        {
+
+        }
+
+        //protected ConfirmPrompt(IModel model, PromptValidator<ConfirmResult> validator = null)
+        //    : base(validator)
+        //{
+        //    this._model = model;
+        //}
+
+        public override async Task<ConfirmResult> Recognize(IBotContext context)
+        {
+            BotAssert.ContextNotNull(context);
+            BotAssert.ActivityNotNull(context.Request);
+            if (context.Request.Type != ActivityTypes.Message)
+                throw new InvalidOperationException("No Message to Recognize");
+
+            IMessageActivity message = context.Request.AsMessageActivity();
+            var confirmResult = new ConfirmResult();
+            Match yesMatch = yes.Match(message.Text);
+            Match noMatch = no.Match(message.Text);
+            if (yesMatch.Success)
+            {
+                confirmResult.Status = RecognitionStatus.Recognized;
+                confirmResult.Confirmation = true;
+                confirmResult.Text = yesMatch.Value;
+                await Validate(context, confirmResult);
+            }
+            else if (noMatch.Success)
+            {
+                confirmResult.Status = RecognitionStatus.Recognized;
+                confirmResult.Confirmation = false;
+                confirmResult.Text = noMatch.Value;
+                await Validate(context, confirmResult);
+            }
+            return confirmResult;
```

Comment:
>return confirmResult; [](start = 12, length = 21)

You should ALWAYS call validate (even when not recognized) because you want to give the custom validator a chance to run on every turn of conversation.  This lets the validator not just add validations but also send the re-prompt text as well.  Additionally, the validator may decide to return some default value after so many attempts or who knows what.